site/content/en/contributions/design/wasm-extension.md

```diff
@@ -0,0 +1,54 @@
+---
+title: "Wasm OCI Image Support"
+---
+
+## Motivation
+Envoy Gateway (EG) should support Wasm OCI image as a remote wasm code source.
+This feature will allow users to deploy Wasm modules from OCI registries, such as Docker Hub, 
+Google Container Registry, and Amazon Elastic Container Registry, to Envoy proxies managed by EG.
+
+## Goals
+* Define the system components needed to support Wasm OCI images as remote Wasm code sources.
+
+## Architecture
+
+![](../wasm-extension.png)
+
+### Control Plane Wasm File Cache
+
+Envoy lacks native OCI image support, therefore, EG needs to download Wasm modules from OIC registries, 
+cache them locally in the file system, and serve them to Envoy over HTTP.
+
+**HTTP Code Source:** For HTTP code source, we have two options: serve Wasm modules directly from their 
+original HTTP URLs, or cache them in EG (as with OCI images). Caching both the HTTP Wasm modules and OCI 
+images inside EG can make UI consistent, for example, sha256sum can be calculated on the EG side and made 
+optional in the API. EG can also periodically pull Wasm modules and update the local file cache.
+
```

Comment:
how would resource isolation/authorization work ? i.e. how will EG make sure its not serving a private OCI image with private creds to another proxy or route filter ?

Reply:
im referring to authz here
1. `user1` added a image cred as a `Secret` in `ns1`, and linked this in the `EEP` config for a specific `oci1` private URL targeting `httproute1`
2.  how will EG ensure `user2` created EEP in `ns2` cannot access the same `oci1` from the EG cache

Reply:
We make sure the credential within that EEP wasm setting is correct. In this case:

If `EEP1` in `ns1` created by `user1` and `EEP2` in `ns2` created by `user2` refer to the same OCI image, `user1` need to create `secret1` in `ns1` and `user2` need to `secret2` in `ns2`, both with the correct credential to access to the registry.